presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -148,6 +151,24 @@ public PlanWithProperties visitPlan(PlanNode node, StreamPreferredProperties par
                     parentPreferences.withDefaultParallelism(session));
         }
 
+        @Override
+        public PlanWithProperties visitProject(ProjectNode node, StreamPreferredProperties parentPreferences)
```

Comment:
Why don't we add parallelism for filter/project?

Reply:
If we add an exchange to parallelize here, then you can't remove a sort at the end even if the data is already sorted before the project.  I added this here to enable the optimization for removing sorts on already sorted data, even if there is a projection or filter.

This change affects everything, though, so it does have some risk.  My thinking was that it's  not clear we get a ton of benefit by adding the parallelism if the parents are single stream because we'll bottleneck at the parent node anyway.  However, there could be cases that are negatively impacted.  If you think it's better,  I can remove this commit, and we can just have a less optimal plan for the unpartitioned window function + project + order by case.

Reply:
It could be that source operator releases all rows at once, and parallelism can help speed up the costlly projections/filters. It would be much safer to do this just for window+project case

Reply:
it hard to do for window + project case without pulling this out to its own optimizer, which is waiting on the constraints framework to be in a better state to be extended.  For now, I'll just fix the bug (keep the final sort for window + project since project is going to be parallelized), and when we pull this out, we can optimizer further.